src/main/java/seedu/anilist/logic/parser/ParserUtil.java

```diff
@@ -29,9 +30,13 @@
      */
     public static Index parseIndex(String oneBasedIndex) throws ParseException {
         String trimmedIndex = oneBasedIndex.trim();
-        if (!StringUtil.isNonZeroUnsignedInteger(trimmedIndex)) {
+        if (!StringUtil.isValidNumber(trimmedIndex)) {
             throw new ParseException(MESSAGE_INVALID_INDEX);
         }
+
+        if (!StringUtil.isNonZeroUnsignedInteger(trimmedIndex)) {
+            throw new IntegerOutOfRangeException(1, Integer.MAX_VALUE);
```

Comment:
uhh why does IntegerOutOfRangeException take in 1 and maxint as its parameters? is this necessary?

Reply:
Not exactly, I made it support any range from before, so I can modify it if I just want 1-99999 for example, but I'm not too sure if it's a good idea.